dlt/extract/resource.py

```diff
@@ -480,10 +503,39 @@ def _set_hints(
             if table_schema_template.get("validator") is not None:
                 self.validator = table_schema_template["validator"]
 
+    def compute_table_schema(self, item: TDataItem = None, meta: Any = None) -> TTableSchema:
+        table_schema = super().compute_table_schema(item, meta)
+        if self.incremental and "incremental" not in table_schema:
+            incremental = self.incremental
+            if isinstance(incremental, IncrementalResourceWrapper):
+                incremental = incremental.incremental  # type: ignore[assignment]
+            if incremental:
+                table_schema["incremental"] = incremental.to_table_hint()  # type: ignore[attr-defined]
+
+        return table_schema
+
     def bind(self: TDltResourceImpl, *args: Any, **kwargs: Any) -> TDltResourceImpl:
         """Binds the parametrized resource to passed arguments. Modifies resource pipe in place. Does not evaluate generators or iterators."""
         if self._args_bound:
             raise TypeError(f"Parametrized resource {self.name} is not callable")
+
+        incremental = self.incremental
+
+        if self._initial_incremental and (
```

Comment:
why we need to do it here? isn't it enough if we keep Incremental in thints (via update_hints?)